yarn-project/simulator/src/client/execution_result.ts

```diff
@@ -104,7 +115,8 @@ function collectNoteEncryptedLogs(execResult: ExecutionResult): CountedLog<Encry
  * @returns All encrypted logs.
  */
 export function collectSortedNoteEncryptedLogs(execResult: ExecutionResult): EncryptedNoteFunctionL2Logs {
-  const allLogs = collectNoteEncryptedLogs(execResult);
+  const nullifiedNoteHashCounters = Array.from(collectNullifiedNoteHashCounters(execResult).keys()).flat();
```

Comment:
Will be more efficient just leave it as a map. `filter(includes)` is almost like nested loops.